### PR TITLE
bug: failed to create cluster with proxy

### LINKED ISSUE #2633: kinder: failed to create cluster with proxy
## Is this a BUG REPORT or FEATURE REQUEST?

Choose one: BUG REPORT

## Versions

**kinder version** (use `kinder version`): `kinder version: 0.1.0-alpha.3 (commit hash: fc9ecec9b8e3f42e59b9183f7bf0d22915252e4b)`

**Environment**:
- **OS** (e.g. from /etc/os-release): `Ubuntu 20.04.3 LTS (Focal Fossa)`
- **Kernel** (e.g. `uname -a`): `Linux k8s 5.4.0-92-generic #103-Ubuntu SMP Fri Nov 26 16:15:10 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux`
- **Container runtime (CRI)** (e.g. containerd, cri-o): `docker`

## What happened?

Failed to create cluster with proxy.

```bash
ubuntu@k8s:~$ HTTP_PROXY=http://192.168.64.1:1087 HTTPS_PROXY=http://192.168.64.1:1087 ./kinder create cluster --control-plane-nodes 1 --worker-nodes 1 --image kindest/node:v1.23.1 --loglevel trace
DEBU[10:48:14] Running: [docker ps -q -a --no-trunc --filter label=io.k8s.sigs.kind.cluster --format {{.Label "io.k8s.sigs.kind.cluster"}}] 
Creating cluster "kind" ...
Ensuring node image (kindest/node:v1.23.1) 🖼
DEBU[10:48:14] Running: [docker pull kindest/node:v1.23.1]  
Preparing nodes 📦📦
DEBU[10:48:17] Running: [docker run -d --entrypoint=sleep --name=kind-detect-9a1170b9-87bb-46fc-9100-29a2cad12761 kindest/node:v1.23.1 infinity] 
DEBU[10:48:18] Running: [docker exec kind-detect-9a1170b9-87bb-46fc-9100-29a2cad12761 /bin/sh -c which docker || true] 
DEBU[10:48:18] Running: [docker rm -f kind-detect-9a1170b9-87bb-46fc-9100-29a2cad12761] 
INFO[10:48:18] Detected containerd container runtime for image kindest/node:v1.23.1 
INFO[10:48:18] Creating nodes...                            
DEBU[10:48:18] Running: [docker network inspect -f {{range (index (index . "IPAM") "Config")}}{{index . "Subnet"}} {{end}} bridge] 
DEBU[10:48:18] Running: [docker network inspect -f {{range (index (index . "IPAM") "Config")}}{{index . "Subnet"}} {{end}} bridge] 
DEBU[10:48:18] Running: [docker info --format '{{json .SecurityOptions}}'] 
DEBU[10:48:18] Running: [docker info --format '{{json .SecurityOptions}}'] 
DEBU[10:48:18] Running: [docker run --detach --tty --label io.x-k8s.kind.cluster=kind --label io.k8s.sigs.kind.cluster=kind --hostname kind-worker-1 --name kind-worker-1 --label io.x-k8s.kind.role=worker --label io.k8s.sigs.kind.role=worker -e https_proxy=http://192.168.64.1:1087 -e = -e NO_PROXY=172.17.0.0/16, -e no_proxy=172.17.0.0/16, -e HTTP_PROXY=http://192.168.64.1:1087 -e http_proxy=http://192.168.64.1:1087 -e HTTPS_PROXY=http://192.168.64.1:1087 --privileged --security-opt seccomp=unconfined --tmpfs /tmp --tmpfs /run --volume /var --volume /lib/modules:/lib/modules:ro kindest/node:v1.23.1] 
DEBU[10:48:18] Running: [docker run --detach --tty --label io.x-k8s.kind.cluster=kind --label io.k8s.sigs.kind.cluster=kind --hostname kind-control-plane-1 --name kind-control-plane-1 --label io.x-k8s.kind.role=control-plane --label io.k8s.sigs.kind.role=control-plane -e HTTPS_PROXY=http://192.168.64.1:1087 -e https_proxy=http://192.168.64.1:1087 -e = -e NO_PROXY=172.17.0.0/16, -e no_proxy=172.17.0.0/16, -e HTTP_PROXY=http://192.168.64.1:1087 -e http_proxy=http://192.168.64.1:1087 --privileged --security-opt seccomp=unconfined --tmpfs /tmp --tmpfs /run --volume /var --volume /lib/modules:/lib/modules:ro --publish=46831:6443/TCP kindest/node:v1.23.1] 
DEBU[10:48:18] Reading container list for cluster kind      
DEBU[10:48:18] Running: [docker ps -q -a --no-trunc --filter label=io.k8s.sigs.kind.cluster=kind --format {{.Names}}] 
ERRO[10:48:18] exit status 125                              
Error: failed to create cluster: exit status 125
ubuntu@k8s:~$ 
```

## What you expected to happen?

Success to create cluster with proxy.

## How to reproduce it (as minimally and precisely as possible)?

```bash
HTTP_PROXY=http://192.168.64.1:1087 HTTPS_PROXY=http://192.168.64.1:1087 ./kinder create cluster --control-plane-nodes 1 --worker-nodes 1 --image kindest/node:v1.23.1
```

### PR DESCRIPTION
**Pull request by PeerXu**
Fix #2633

### CODE REVIEW

File: kinder/pkg/cri/nodes/common/util.go [hunk 1]
```diff
@@ -101,13 +101,13 @@ func getProxyEnvs() (map[string]string, error) {
 	}
 	// Specifically add the cluster subnets to NO_PROXY if we are using a proxy
 	if len(envs) > 0 {
-		noProxy := envs[noProxy]
-		if noProxy != "" {
-			noProxy += ","
+		noProxyVal := envs[noProxy]
```
Comment (SataQiu) [inline kinder/pkg/cri/nodes/common/util.go:104]: Good catch!

**Review by SataQiu — COMMENTED** [review]
/approve

cc @neolit123

**Review by neolit123 — COMMENTED** [review]
/lgtm
Thanks

**Pull request merged**